src/Microsoft.DotNet.VersionTools/Automation/GitHubApi/GitHubClient.cs

```diff
@@ -30,18 +36,21 @@ public class GitHubClient : IDisposable
             "X-RateLimit-Reset"
         };
 
-        private GitHubAuth _auth;
-
         private HttpClient _httpClient;
 
+        public GitHubAuth Auth { get; }
+
         public GitHubClient(GitHubAuth auth)
         {
-            _auth = auth;
+            Auth = auth;
 
             _httpClient = new HttpClient();
             _httpClient.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
-            _httpClient.DefaultRequestHeaders.Add("Authorization", $"token {auth.AuthToken}");
-            _httpClient.DefaultRequestHeaders.Add("User-Agent", auth.User);
+            _httpClient.DefaultRequestHeaders.Add("User-Agent", auth?.User ?? DefaultUserAgent);
+            if (auth?.AuthToken != null)
```

Comment:
Should all the usages of `Auth` be updated to allow for `null`?

Reply:
There are 3 other usages I can find, and I don't think `null` makes sense for them (but I do think they deserve handling):
 * 2 in `PutGitHubFileAsync`, which I think I should add an early descriptive exception for to be clear it's not possible anonymously. (And I think I should add that to other methods that depend on the headers being set too.)
 * 1 passing `client.Auth` when creating a `PullRequestCreator` in `UpdateToRemoteDependenciesAndSubmitPullRequest`. `PullRequestCreator` goes on to use `.User` and `.Email`. The build task would have already failed before this point, but might as well guard against it in the `PullRequestCreator` constructor to show intent and for external use.